pkg/queuecontroller/controllers/queue_controller.go

```diff
@@ -54,7 +55,12 @@ func (r *QueueReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl
 	queue := &v2.Queue{}
 	err := r.Get(ctx, req.NamespacedName, queue)
 	if err != nil {
-		return ctrl.Result{}, client.IgnoreNotFound(err)
+		ignoreNotFoundErr := client.IgnoreNotFound(err)
+		if ignoreNotFoundErr == nil {
+			// If the queue is not found, reset its metrics
+			metrics.ResetQueueMetrics(req.Name)
```

Comment:
Even if the get fails only once? occasional error can occur if the API server is under pressure

Reply:
I do it only for "NotFound" error, not on every error, is the API server has some kind of issue it would return a different error, and not "NotFound" error